src/java.base/share/classes/sun/security/ssl/CertificateRequest.java

```diff
@@ -128,6 +128,10 @@ private static String[] getKeyTypes(byte[] ids) {
             ArrayList<String> keyTypes = new ArrayList<>(3);
             for (byte id : ids) {
                 ClientCertificateType cct = ClientCertificateType.valueOf(id);
+                if(cct == null) {
+                    throw new IllegalArgumentException(id + " was " +
+                            "not a valid ClientCertificateType id");
+                }
```

Comment:
It may not comply to TLS specification if throwing exception here.  Unknown types should be ignored for compatibility.
- if (cct.isAvailable) {
+ if (cct != null && cct.isAvailable) {